### PR TITLE
Replace PHPUnit_Framework_MockObject_MockObject with newer class

### PR DESCRIPTION
**Pull request by hws47a**
### Description (*)
Replace all usages of outdated `PHPUnit_Framework_MockObject_MockObject` with 
`\PHPUnit\Framework\MockObject\MockObject`

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by lbajsarowicz** [comment]
@hws47a Great idea to change that. Please keep in mind that you'll have to fix Static Tests eventually.

**Comment by hws47a** [comment]
@lbajsarowicz thanks. Yeah, just thought about it. If I will bulk update all of the unit tests, they'll probably fail all together in static tests.

**Comment by lbajsarowicz** [comment]
@hws47a Wow. The only tests failed are Functional Tests (false negative). I've made them run again and then it should pass. Thanks for your contribution. I'm going to approve PR.

**Review by lbajsarowicz — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @lbajsarowicz, thank you for the review.
[ENGCOM-6831](https://jira.corp.magento.com/browse/ENGCOM-6831) has been created to process this Pull Request
:eight_spoked_asterisk: @lbajsarowicz, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by hws47a** [comment]
@lbajsarowicz yeah, I saw it. And it's a little bit too early to approve. It's a draft and WIP.

**Comment by hws47a** [comment]
Hi @lbajsarowicz, it's now ready. 
And it seems there are no static check errors at all.

File: lib/internal/Magento/Framework/App/Test/Unit/Language/DictionaryTest.php [hunk 1]
```diff
@@ -17,17 +17,17 @@ class DictionaryTest extends \PHPUnit\Framework\TestCase
     private $model;
 
     /**
-     * @var \PHPUnit_Framework_MockObject_MockObject
+     * @var \PHPUnit\Framework\MockObject\MockObject
```
Comment (lbajsarowicz) [inline lib/internal/Magento/Framework/App/Test/Unit/Language/DictionaryTest.php:20]: Could you also provide `|ReadFactory`?

File: lib/internal/Magento/Framework/App/Test/Unit/Language/DictionaryTest.php [hunk 2]
```diff
@@ -17,17 +17,17 @@ class DictionaryTest extends \PHPUnit\Framework\TestCase
...
      */
     private $readFactory;
 
     /**
-     * @var \PHPUnit_Framework_MockObject_MockObject
+     * @var \PHPUnit\Framework\MockObject\MockObject
```
Comment (lbajsarowicz) [inline lib/internal/Magento/Framework/App/Test/Unit/Language/DictionaryTest.php:25]: `|ComponentRegistrar`

File: lib/internal/Magento/Framework/App/Test/Unit/Language/DictionaryTest.php [hunk 3]
```diff
@@ -17,17 +17,17 @@ class DictionaryTest extends \PHPUnit\Framework\TestCase
...
      */
     private $componentRegistrar;
 
     /**
-     * @var \PHPUnit_Framework_MockObject_MockObject
+     * @var \PHPUnit\Framework\MockObject\MockObject
```
Comment (lbajsarowicz) [inline lib/internal/Magento/Framework/App/Test/Unit/Language/DictionaryTest.php:30]: `|ConfigFactory`

File: lib/internal/Magento/Framework/Acl/Test/Unit/AclResource/ProviderTest.php [hunk 4]
```diff
@@ -13,22 +13,22 @@ class ProviderTest extends \PHPUnit\Framework\TestCase
     protected $_model;
 
     /**
-     * @var \PHPUnit_Framework_MockObject_MockObject
+     * @var \PHPUnit\Framework\MockObject\MockObject
```
Comment (lbajsarowicz) [inline lib/internal/Magento/Framework/Acl/Test/Unit/AclResource/ProviderTest.php:16]: `|ConfigReader`

File: lib/internal/Magento/Framework/Acl/Test/Unit/AclResource/ProviderTest.php [hunk 5]
```diff
@@ -13,22 +13,22 @@ class ProviderTest extends \PHPUnit\Framework\TestCase
...
      */
     protected $_configReaderMock;
 
     /**
-     * @var \PHPUnit_Framework_MockObject_MockObject
+     * @var \PHPUnit\Framework\MockObject\MockObject
```
Comment (lbajsarowicz) [inline lib/internal/Magento/Framework/Acl/Test/Unit/AclResource/ProviderTest.php:21]: `|TreeBuilder`

File: lib/internal/Magento/Framework/Acl/Test/Unit/BuilderTest.php [hunk 6]
```diff
@@ -10,27 +10,27 @@
 class BuilderTest extends \PHPUnit\Framework\TestCase
 {
     /**
-     * @var \PHPUnit_Framework_MockObject_MockObject
+     * @var \PHPUnit\Framework\MockObject\MockObject
```
Comment (lbajsarowicz) [inline lib/internal/Magento/Framework/Acl/Test/Unit/BuilderTest.php:13]: Could you add `|` and the accurate class reference?

File: lib/internal/Magento/Framework/Acl/Test/Unit/BuilderTest.php [hunk 7]
```diff
@@ -10,27 +10,27 @@
...
      */
     protected $_aclFactoryMock;
 
     /**
-     * @var \PHPUnit_Framework_MockObject_MockObject
+     * @var \PHPUnit\Framework\MockObject\MockObject
```
Comment (lbajsarowicz) [inline lib/internal/Magento/Framework/Acl/Test/Unit/BuilderTest.php:18]: Could you add `|` and the accurate class reference?

File: lib/internal/Magento/Framework/Acl/Test/Unit/BuilderTest.php [hunk 8]
```diff
@@ -10,27 +10,27 @@
...
      */
     protected $_aclMock;
 
     /**
-     * @var \PHPUnit_Framework_MockObject_MockObject
+     * @var \PHPUnit\Framework\MockObject\MockObject
```
Comment (lbajsarowicz) [inline lib/internal/Magento/Framework/Acl/Test/Unit/BuilderTest.php:23]: Could you add `|` and the accurate class reference?

File: lib/internal/Magento/Framework/Acl/Test/Unit/BuilderTest.php [hunk 9]
```diff
@@ -10,27 +10,27 @@
...
      */
     protected $_ruleLoader;
 
     /**
-     * @var \PHPUnit_Framework_MockObject_MockObject
+     * @var \PHPUnit\Framework\MockObject\MockObject
```
Comment (lbajsarowicz) [inline lib/internal/Magento/Framework/Acl/Test/Unit/BuilderTest.php:28]: Could you add `|` and the accurate class reference?

File: lib/internal/Magento/Framework/Acl/Test/Unit/BuilderTest.php [hunk 10]
```diff
@@ -10,27 +10,27 @@
...
      */
     protected $_roleLoader;
 
     /**
-     * @var \PHPUnit_Framework_MockObject_MockObject
+     * @var \PHPUnit\Framework\MockObject\MockObject
```
Comment (lbajsarowicz) [inline lib/internal/Magento/Framework/Acl/Test/Unit/BuilderTest.php:33]: Could you add `|` and the accurate class reference?

File: lib/internal/Magento/Framework/Amqp/Test/Unit/ConfigTest.php [hunk 11]
```diff
@@ -12,7 +12,7 @@
 class ConfigTest extends \PHPUnit\Framework\TestCase
 {
     /**
-     * @var \PHPUnit_Framework_MockObject_MockObject
+     * @var \PHPUnit\Framework\MockObject\MockObject
```
Comment (lbajsarowicz) [inline lib/internal/Magento/Framework/Amqp/Test/Unit/ConfigTest.php:15]: Could you add `|` and the accurate class reference?

File: lib/internal/Magento/Framework/Api/Test/Unit/Code/Generator/EntityChildTestAbstract.php [hunk 12]
```diff
@@ -14,7 +14,7 @@
 abstract class EntityChildTestAbstract extends \PHPUnit\Framework\TestCase
 {
     /**
-     * @var Io | \PHPUnit_Framework_MockObject_MockObject
+     * @var Io | \PHPUnit\Framework\MockObject\MockObject
```
Comment (lbajsarowicz) [inline lib/internal/Magento/Framework/Api/Test/Unit/Code/Generator/EntityChildTestAbstract.php]: Remove space between `|`

File: lib/internal/Magento/Framework/Api/Test/Unit/Code/Generator/EntityChildTestAbstract.php [hunk 13]
```diff
@@ -24,11 +24,11 @@ abstract class EntityChildTestAbstract extends \PHPUnit\Framework\TestCase
     protected $generator;
 
     /**
-     * @var \Magento\Framework\Code\Generator\ClassGenerator | \PHPUnit_Framework_MockObject_MockObject
+     * @var \Magento\Framework\Code\Generator\ClassGenerator | \PHPUnit\Framework\MockObject\MockObject
```
Comment (lbajsarowicz) [inline lib/internal/Magento/Framework/Api/Test/Unit/Code/Generator/EntityChildTestAbstract.php]: Remove space around `|`

File: lib/internal/Magento/Framework/Api/Test/Unit/Code/Generator/EntityChildTestAbstract.php [hunk 14]
```diff
@@ -24,11 +24,11 @@ abstract class EntityChildTestAbstract extends \PHPUnit\Framework\TestCase
...
      */
     protected $classGenerator;
 
-    /** @var \PHPUnit_Framework_MockObject_MockObject | \Magento\Framework\Code\Generator\DefinedClasses */
+    /** @var \PHPUnit\Framework\MockObject\MockObject | \Magento\Framework\Code\Generator\DefinedClasses */
```
Comment (lbajsarowicz) [inline lib/internal/Magento/Framework/Api/Test/Unit/Code/Generator/EntityChildTestAbstract.php]: Remove space around `|`

File: lib/internal/Magento/Framework/Api/Test/Unit/Code/Generator/GenerateMapperTest.php [hunk 15]
```diff
@@ -11,7 +11,7 @@
 class GenerateMapperTest extends \PHPUnit\Framework\TestCase
 {
     /**
-     * @var \PHPUnit_Framework_MockObject_MockObject
+     * @var \PHPUnit\Framework\MockObject\MockObject
```
Comment (lbajsarowicz) [inline lib/internal/Magento/Framework/Api/Test/Unit/Code/Generator/GenerateMapperTest.php:14]: Could you add `|` and the accurate class reference?

File: lib/internal/Magento/Framework/Api/Test/Unit/Code/Generator/GenerateSearchResultsTest.php [hunk 16]
```diff
@@ -11,7 +11,7 @@
 class GenerateSearchResultsTest extends \PHPUnit\Framework\TestCase
 {
     /**
-     * @var \PHPUnit_Framework_MockObject_MockObject
+     * @var \PHPUnit\Framework\MockObject\MockObject
```
Comment (lbajsarowicz) [inline lib/internal/Magento/Framework/Api/Test/Unit/Code/Generator/GenerateSearchResultsTest.php:14]: Could you add `|` and the accurate class reference?

File: lib/internal/Magento/Framework/App/Test/Unit/ActionFlagTest.php [hunk 17]
```diff
@@ -13,7 +13,7 @@ class ActionFlagTest extends \PHPUnit\Framework\TestCase
     protected $_actionFlag;
 
     /**
-     * @var \PHPUnit_Framework_MockObject_MockObject
+     * @var \PHPUnit\Framework\MockObject\MockObject
```
Comment (lbajsarowicz) [inline lib/internal/Magento/Framework/App/Test/Unit/ActionFlagTest.php:16]: Could you add `|` and the accurate class reference?

File: lib/internal/Magento/Framework/App/Test/Unit/AreaListTest.php [hunk 18]
```diff
@@ -21,7 +21,7 @@ class AreaListTest extends \PHPUnit\Framework\TestCase
     protected $_resolverFactory;
 
     /**
-     * @var \PHPUnit_Framework_MockObject_MockObject
+     * @var \PHPUnit\Framework\MockObject\MockObject
```
Comment (lbajsarowicz) [inline lib/internal/Magento/Framework/App/Test/Unit/AreaListTest.php:24]: Could you add `|` and the accurate class reference?

File: lib/internal/Magento/Framework/App/Test/Unit/AreaListTest.php [hunk 19]
```diff
@@ -145,7 +145,7 @@ public function testGetArea()
     }
 
     /**
-     * @return \PHPUnit_Framework_MockObject_MockObject
+     * @return \PHPUnit\Framework\MockObject\MockObject
```
Comment (lbajsarowicz) [inline lib/internal/Magento/Framework/App/Test/Unit/AreaListTest.php:148]: Could you add `|` and the accurate class reference?

File: lib/internal/Magento/Framework/App/Test/Unit/AreaTest.php [hunk 20]
```diff
@@ -21,42 +21,42 @@ class AreaTest extends \PHPUnit\Framework\TestCase
     protected $objectManager;
 
     /**
-     * @var \Magento\Framework\Event\ManagerInterface | \PHPUnit_Framework_MockObject_MockObject
+     * @var \Magento\Framework\Event\ManagerInterface | \PHPUnit\Framework\MockObject\MockObject
```
Comment (lbajsarowicz) [inline lib/internal/Magento/Framework/App/Test/Unit/AreaTest.php]: Could you remove space around `|`

File: lib/internal/Magento/Framework/App/Test/Unit/AreaTest.php [hunk 21]
```diff
@@ -21,42 +21,42 @@ class AreaTest extends \PHPUnit\Framework\TestCase
...
      */
     protected $eventManagerMock;
 
     /**
-     * @var \Magento\Framework\ObjectManagerInterface | \PHPUnit_Framework_MockObject_MockObject
+     * @var \Magento\Framework\ObjectManagerInterface | \PHPUnit\Framework\MockObject\MockObject
```
Comment (lbajsarowicz) [inline lib/internal/Magento/Framework/App/Test/Unit/AreaTest.php]: Could you remove space around `|`

**Review by lbajsarowicz — CHANGES_REQUESTED** [review]
Could you remove spaces around `|` in all modified files?
Also - we need the real class reference next to `MockObject` that is missing in some cases

**Comment by hws47a** [comment]
@lbajsarowicz have you seen the number of files changed? :)
I have no intension to fix anything else in this PR except just replacing outdated class name.

**Comment by lbajsarowicz** [comment]
Remove just spaces around `|` as it's just another smell (that should be detected by Static tests actually). Adding missing annotations for original classes could be done in separate PR.

**Comment by hws47a** [comment]
> Remove just spaces around `|` as it's just another smell (that should be detected by Static tests actually). Adding missing annotations for original classes could be done in separate PR.

@lbajsarowicz  Done

**Comment by hws47a** [comment]
I already had to fix two conflicts during rebase, so it's better to merge it as soon as possible

**Comment by ihor-sviziev** [comment]
@hws47a wow. You did really huge work!
Unfortunately such bug PRs is really hard to merge as they all the time has conflicts with other PRs. I think it’s better to split the work to smaller parts that could be quite easy tested and quite quickly delivered.

Am I correct @slavvka @buskamuza @VladimirZaets ?

**Comment by lbajsarowicz** [comment]
That can be delivered the same way that I delivered MFTF changes (1 per module)

**Comment by hws47a** [comment]
@lbajsarowicz @ihor-sviziev Thank you for suggestion.
I'll do it.

**Comment by hws47a** [comment]
@lbajsarowicz @ihor-sviziev I've create one, let's test-drive on it.

**Comment by ihor-sviziev** [comment]
Hi @hws47a,
This PR already have some conflicts as I said above, I think we can close it, as you’re going to create separate PRs for each module.

**Pull request closed without merging**